float64 time
int32 x_position
int32 y_position
int32 z_position